src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -941,7 +927,7 @@ protected void doRun() throws Exception {
     public SnapshotIndexCommit snapshotIndex() throws EngineException {
         // we have to flush outside of the readlock otherwise we might have a problem upgrading
         // the to a write lock when we fail the engine in this operation
-        flush(FlushType.COMMIT, false, true);
+        flush(false, false, true);
```

Comment:
This is the only place that commits w/o pinching the xlog right?  Why do we do this?  Is it so the snapshot won't fail when a recovery is in process?

I think somehow we should make our xlog reference counted in the future, so that a recovery can "incref" it to prevent deletion before the point-in-time it reserved, and then flush doesn't need this boolean param anymore: it would just decref its reference.  Later ...